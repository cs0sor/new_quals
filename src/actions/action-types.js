export const ADD_QUALIFICATION = 'ADD_QUALIFICATION'
export const ADD_UNIT_TO_QUALIFICATION = 'ADD_UNIT_TO_QUALIFICATION'

