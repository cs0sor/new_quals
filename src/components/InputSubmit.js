import FormControl from "react-bootstrap/lib/FormControl"
import Button from 'react-bootstrap/lib/Button'
import React from "react"

export default class InputSubmit extends React.Component {
    
    constructor(props) {
        super(props)
        this.state = { title: ''}
        this.handleUpdate = this.handleUpdate.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleUpdate(e) {
        this.setState({title: e.target.value})
    }

    handleSubmit(e) {
        this.props.submitAction(this.state.title)
    }

    render() {
        return (
            <div>
            <FormControl
                componentClass='input'
                placeholder='qualification title'
                onChange={e => this.handleUpdate(e)}
                value={this.state.title}
            />
            <Button onClick={ e => this.handleSubmit(e)}>
                Submit
            </Button>
            </div>
        )
    }
}