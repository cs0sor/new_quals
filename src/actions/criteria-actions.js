import * as types from './action-types'

export const splitFromCriteria = (criteriaId, groupId) => ({
  type: types.SPLIT_FROM_CRITERIA,
  criteriaId: criteriaId,
  groupId: groupId
})

export const mergeGroup = (payload) => ({
    type: types.MERGE_GROUP,
    payload: payload
})

export const updateMinScore = (payload) => ({
    type: types.UPDATE_MIN_SCORE,
    payload: payload
})

export const changeScoreCriteria = (payload) => ({
    type: types.CHANGE_SCORE_CRITERIA,
    payload: payload
})