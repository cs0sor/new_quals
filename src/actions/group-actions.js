import * as types from './action-types'

export const addNewGroup = () => ({
  type: types.ADD_NEW_GROUP
})