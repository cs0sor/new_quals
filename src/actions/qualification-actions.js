import * as types from './action-types'

export const addQualification = (title) => {
  return {
    type: types.ADD_QUALIFICATION,
    title
  }
}

export const addUnitToQualification = (selected) => {
  return {
    type: types.ADD_UNIT_TO_QUALIFICATION,
    selected
  }
}
