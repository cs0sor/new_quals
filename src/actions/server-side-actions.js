import * as types from './action-types'
import FuzzySearch from '../search'

const qualificationSaved = () => (
  {type: types.QUALIFICATION_SAVED}
)

export function saveQualification(payload) {
  return function(dispatch) {
    fetch('/qualification.json/put.json', {
        method: 'post',
        body: JSON.stringify(payload),
      }).then(response => response.json()
    ).then(payload => {
      dispatch(qualificationSaved())
    }).catch(err => {
      console.log(err);
    });
    return false
  }
}


const setQualification = (payload) =>
  ({
    type: types.SET_QUALIFICATION_STATE,
    payload: {
      ...payload,
      searchResults: Object.keys(payload.units),
      dataLoaded: true}
  })

function configureData(payload) {
  return {
    availableQuals: {
      [payload.qualification.id]:
        {
          id: payload.qualification.id,
          title: payload.qualification.title,
          live: payload.qualification.live,
        }
      },
    selectedQual: payload.qualification.id,
    selectedGroup: null,
    saveable: false,
    searchResults: [],
    units: payload.units,
    groups: payload.groups,
    criteria: payload.criteria,
  }
}

export function getQualification(qualId) {
  return function(dispatch) {
    fetch('/qualification.json/get.json', {
        method: 'post',
        body: JSON.stringify({qual_id: qualId}),
      }).then(response => response.json()
    ).then(payload => {
      const fs = FuzzySearch
      fs.init(payload.units)
      dispatch(setQualification({...configureData(payload), dataLoaded: true}))
    }).catch(err => {
      console.log(err);
    });
    return false
  }
}
