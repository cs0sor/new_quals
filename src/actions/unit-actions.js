import * as types from './action-types'

export const addUnitToGroup = (selected) => ({
  type: types.ADD_UNIT_TO_GROUP,
  selected
})

export const removeUnitFromGroup = (selected) => ({
  type: types.REMOVE_UNIT_FROM_GROUP,
  selected
})

export const searchUnits = (term) => ({
  type: types.SEARCH_UNITS,
  term
})

