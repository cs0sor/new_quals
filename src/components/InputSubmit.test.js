import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { mount } from 'enzyme'
import configureStore from '../store/configure-store'
import InputSubmit from '../components/InputSubmit'

import { addQualification } from '../actions/qualification-actions'

Enzyme.configure({ adapter: new Adapter() })

it('renders the input and submit component', () => {
	const component = renderer.create(<InputSubmit />)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})

it('checks to see if text is rendered by input', () => {
	const component = mount(<InputSubmit id="input-contol" />)

	const input = component.find('input')

	input.simulate('change', { target: { value: 'A New Qualification' } })
	expect(component.state().title).toEqual('A New Qualification')
})

it('check action button submit', () => {
	const store = configureStore()

	const addQuestion = (e) => {
		store.dispatch(addQualification(e))
	}

	const component = mount(<InputSubmit id="input-contol" store={store} submitAction={addQuestion} />)

	const input = component.find('input')
	input.simulate('change', { target: { value: 'A New Qualification' } })

	const button = component.find('button')
	button.simulate('click')

	expect(store.getState().qualReducer.availableQuals[3] === 'A New Qualification')
})
