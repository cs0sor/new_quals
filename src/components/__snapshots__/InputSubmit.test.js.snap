// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the input and submit component 1`] = `
<div>
  <input
    className="form-control"
    id={undefined}
    onChange={[Function]}
    placeholder="qualification title"
    type={undefined}
    value=""
  />
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;
